test/01-smoke.test.js

```diff
@@ -38,17 +38,25 @@ test('smoke', async function ({ ok, is, plan, comment, teardown, timeout, end })
 
   comment('running')
   const link = 'pear://' + key
-  const running = await Helper.open(link, { tags: ['exit'] })
-
-  const { value } = await running.inspector.evaluate('Pear.versions()', { awaitPromise: true })
-  is(value?.app?.key, key, 'app version matches staged key')
-
-  const dhtBootstrap = await running.inspector.evaluate('Pear.config.dht.bootstrap')
-  is(JSON.stringify(dhtBootstrap.value), JSON.stringify(Pear.config.dht.bootstrap), 'dht bootstrap matches Pear.config.dth.bootstrap')
 
-  await running.inspector.close()
-  const { code } = await running.until.exit
-  is(code, 0, 'exit code is 0')
+  const pipe = Pear.worker.run(link)
+  pipe.on('data', (data) => {
+    const obj = JSON.parse(data.toString())
+    if (obj.type === 'versions') {
+      is(obj.value.app.key, key, 'app version matches staged key')
+    } else if (obj.type === 'dhtBootstrap') {
+      is(JSON.stringify(obj.value), JSON.stringify(Pear.config.dht.bootstrap), 'dht bootstrap matches Pear.config.dth.bootstrap')
+    }
+  })
+  pipe.on('end', () => {
+    ok(true, 'pipe ended')
+  })
+
+  pipe.write('versions')
+  await new Promise((resolve) => setTimeout(resolve, 1000))
```

Comment:
I think we should await pipe response instead of using `setTimeout`.

Something like:

```
async request (pipe, msg) => new Promise((resolve) => {
  pipe.on('data', resolve)
  pipe.write(msg)
}

request(worker, 'version')
```

Reply:
updated